tools/test.py

```diff
@@ -1581,18 +1584,33 @@ def Main():
     logger.addHandler(fh)
 
   workspace = abspath(join(dirname(sys.argv[0]), '..'))
-  suites = GetSuites(join(workspace, 'test'))
+  test_root = join(workspace, 'test')
+  suites = GetSuites(test_root)
   repositories = [TestRepository(join(workspace, 'test', name)) for name in suites]
   repositories += [TestRepository(a) for a in options.suite]
 
   root = LiteralTestSuite(repositories)
+  default_suites = DefaultSuites(test_root)
   if len(args) == 0:
-    paths = [SplitPath(t) for t in BUILT_IN_TESTS]
+    paths = default_suites
   else:
     paths = [ ]
     for arg in args:
-      path = SplitPath(NormalizePath(arg))
-      paths.append(path)
+      # if argument is of the form suites_<subsystem> run an
+      # entire set of tests for a subsystem, e.g., suites_http.
+      # "suites_default" indicates that a reasonable default set of
+      # suites should be run.
+      suite_match = suite_regex.match(arg)
+      if suite_match:
+        suite = suite_match.group(1)
+        if suite == 'default':
+          paths += default_suites
+        else:
+          path = SplitPath(NormalizePath("*/test-%s-*" % (suite)))
```

Comment:
I would suggest just using `"*/*%s*"`, so that we get e.g. `test/parallel/test-cluster-dgram-1.js` even if we specify just `dgram`

Reply:
Also, this approach feels a bit like it might be too much magic :) Not sure what to do about that, though.

Reply:
My thinking was that it definitely feels like the right move to add `suites_default`, as it allows folks to stop manually updating a variable in several places whenever a new test folder is added ... and, if we already have this shorthand added, it feels really elegant to be able to also specific a specific suite to run like `http`.

An alternative might be something like `--suites=default`, `--suites=http`?

Reply:
I feel like this is the good kind of magic, all you have to remember is it's basically doing a search for the string you provide right?

As long as it still works with the existing uses:

```bash
tools/test.py parallel
tools/test.py parallel/test-assert
tools/test.py test/parallel/test-assert.js
```

which I think it will, I think this will be a great addition. I'm already looking forward to being able to do `tools/test.py foo`.

Reply:
@gibfahn my concern was if there's a collision on a folder name, e.g., if I want to run `addons`. Perhaps check if the `arg` is in the list of folders, and then assume it's the name of a suite otherwise? I will happily implement whichever magic folks land on ... it would definitely be nice to not have the `suites_` prefix.

Reply:
What you said sounds reasonable, but probably worth waiting to see what others think.

Reply:
@gibfahn @addaleax @refack okay, riffing on @gibfahn's idea what if instead of requiring the `subsystem_` prefix, we did something clever like this:

```python
def MaybeSubsystem(test_root, arg):
  if (arg == 'default'):
    return True
  test_dirs = os.listdir(test_root)
  subsystem_regex = re.compile(r'^[a-zA-Z-]*$')
  return subsystem_regex.match(arg) and (arg not in test_dirs) and (arg not in IGNORED_SUITES)
```

If the `arg` passed to `tool/test.py` matches a test-folder, or contains special characters like `/` or `*` we execute tests like usual. If the `arg` doesn't match a folder, and doesn't contain special characters, assume you're trying to test a subsystem ... at this point you'd be able to:

`python tools/test.py http child-process`

or, you can always use the old approach of:

`python tools/test.py parallel/test-assert`

and things would be have as usual ...

I think this small change would make this a pretty darn slick feature.

Reply:
@bnoordhuis @refack @gibfahn pushed the slight refactor that eliminates the need for a `subsystem_` prefix, it definitely fees much more elegant to me now. It's pretty intuitive that I could write:

`python tools/test.py http2` and get tests for `http2`.